src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Serialization/ObjectReader.cs

```diff
@@ -95,39 +95,40 @@ internal sealed partial class ObjectReader : IDisposable
     }
 
     /// <summary>
-    /// Creates an <see cref="ObjectReader"/> from the provided <paramref name="stream"/>.
```

Comment:
ObjectReader/Writer were tweaked a bit to make them amenable to writing/reading out multiple assets to the same stream.  this involved allowing the 'validation bytes' to not just be added automatically at the start of the stream (and then be checked when reading immediatley from the stream), but instead be written/read based on when the client asks it to.